backend/src/main/java/com/redhat/cloud/notifications/templates/Rhosak.java

```diff
@@ -0,0 +1,92 @@
+package com.redhat.cloud.notifications.templates;
+
+import com.redhat.cloud.notifications.models.EmailSubscriptionType;
+import io.quarkus.qute.CheckedTemplate;
+import io.quarkus.qute.TemplateInstance;
+
+// Name needs to be "Rhosak" to read templates from resources/templates/Rhosak
+public class Rhosak implements EmailTemplate {
+    private static final String SCHEDULED_UPGRADE = "scheduled-upgrade";
+    private static final String DISRUPTION = "disruption";
+    private static final String INSTANCE_CREATED = "instance-created";
+    private static final String INSTANCE_DELETED = "instance-deleted";
+    private static final String ACTION_REQUIRED = "action-required";
+
+    @Override
+    public TemplateInstance getTitle(String eventType, EmailSubscriptionType type) {
+        if (eventType.equals(SCHEDULED_UPGRADE)) {
+            return Templates.scheduledUpgradeTitle();
+        } else if (eventType.equals(DISRUPTION)) {
+            return Templates.serviceDisruptionTitle();
+        } else if (eventType.equals(INSTANCE_CREATED)) {
+            return Templates.instanceCreatedTitle();
+        } else if (eventType.equals(INSTANCE_DELETED)) {
+            return Templates.instanceDeletedTitle();
+        } else if (eventType.equals(ACTION_REQUIRED)) {
+            return Templates.actionRequiredTitle();
+        }
```

Comment:
`eventType` is `null` on daily aggregation. As it aggregates over all the `eventTypes` found.
Since `SCHEDULED_UPGRADE` is only meant to be used for daily aggregations I think it should be changed with something like:

```suggestion
        if (type == EmailSubscriptionType.INSTANT) {
            if (eventType.equals(DISRUPTION)) {
                return Templates.serviceDisruptionTitle();
            } else if (eventType.equals(INSTANCE_CREATED)) {
                return Templates.instanceCreatedTitle();
            } else if (eventType.equals(INSTANCE_DELETED)) {
                return Templates.instanceDeletedTitle();
            } else if (eventType.equals(ACTION_REQUIRED)) {
                return Templates.actionRequiredTitle();
            }
        } else if (type == EmailSubscriptionType.DAILY) {
            return Templates.dailyTitle();
        }
```

The `scheduledUpgradeTitle` template should probably be removed (or renamed) and a template for the daily aggregation should be created/used instead

Reply:
Why the change of template name to be prefixed with `dailyXX`?

Reply:
Because, as we currently have, all the event types are aggregated into a single payload and sent as one daily email.
In this case (daily) we don't have different emails for each event.

The name can be anything you want, I just choose `dailyTitle` as an example.

Reply:
@akoserwal is this what we want? i.e aggregate all daily emails and send them over one email?

Reply:
> Because, as we currently have, all the event types are aggregated into a single payload and sent as one daily email.
> In this case (daily) we don't have different emails for each event.
> 
> The name can be anything you want, I just choose `dailyTitle` as an example.

Thanks @josejulio , let's get a confirmation about the behaviour we want and I'll do the necessary updates. Based on your description of aggregation I am not sure if this is what we want.

Reply:
I've made the change with the daily template.